### ISSUE TITLE
Proprietary copyrighted code

### ISSUE THREAD

**Issue by KyleSanderson** [danielmm8888/TF2Classic#62]
You guys don't own most, if any of the code you're committing. This was essentially stolen and leaked, and you're re-leaking it.

**Comment by The-Yiffy-Fox** [comment]
VALVe hasn't said anything of this project, or any other one using any of the leaked code.

**Comment by 404UserNotFound** [comment]
Such as many Portal 2 mods which may have been added on Github. Regardless, we've been trying to get in touch with Valve since the project started in order to get an OK from them on what we're doing. Since then, YouTuber ValveNewsNetwork did a video on us which in turn caused Kotaku to create an article about TF2C. Valve has to have seen this project by now.

Yes, there's still the issue of redistributing leaked code, but the code itself is from 2008 and does not feature anything that could be seen as detrimental to Team Fortress 2. We are actively avoiding adding in any features that would cause economical issues to Team Fortress 2 (such as not adding hats or charging any money for access to the project).

I know some people may see what we are doing as an affront to Valve and TF2, or as "us trying to piss of Valve" (which is what Dr. Greg House on AlliedMods thinks), but that's quite the opposite.

I was hoping nobody would comment on this until the project leader Danielmm8888 saw your comment Kyle, but since Yiffy Fox has commented, I felt I should clarify some things.

**Comment by KyleSanderson** [comment]
Quite a while ago I asked in regards to (licensing to) fixing cstrike as there are no active maintainers at Valve and there were tons of crashes. The request was shot down because the game is still in production. If you're granted some sort of license, that would be interesting, but to my knowledge TF2 is still their flagship game. Submitting fixes upstream seems to be the only semi-legitimate route for the leak; if you can find the right employee...

Regardless, you still can't publish essentially stolen code.

**Comment by 404UserNotFound** [comment]
Well I'm sitting in the SourceMod IRC and I see Drunken_F00l is in here. Maybe he'd be the guy to talk to? And honestly, if we had to, we'd start from scratch and code the entire game up from nothing and continue using TF2 assets because that's the main advertising point of TF2C. It's TF2 but with new features as well as some old cut content. Someone else could probably describe it better than me.

The issue with starting from scratch is that it would set our development back so far that we'd be better off just giving up on this project. And we don't want to give up on it.

**Comment by KyleSanderson** [comment]
DF works on Steam, not TF.

I'm not usually the guy to fight for IP, especially with the absolutely terrible track record from Valve. You guys can't use stolen code, it totally illegitimises this entire project.

**Comment by 404UserNotFound** [comment]
As well, I'd like to reiterate the Portal 2 mods point. Some Portal 2 mods have been greenlit and all portal 2 mods use leaked source code. They publish essentially stolen code and Valve is ok with it. The common conception on our team right now is that with all the press coverage, Valve has to have seen it by now.

We all know Valve likes to lurk Facepunch (which is where this started several months ago). When the project first started, people were scared about potential legal action. To quell fears, I put out my legitimate email address and a general statement to Valve in the event that they saw what we were doing and read my post, and I stated that if they didn't want us doing this, all they had to do was email me and we'd gladly stop right then and there, no legal action necessary.

I've not received an email yet, and I'm in frequent correspondence with Eric S. of Valve over TF2 bugs (most recently some Watchdog crashes on my Ubuntu 12.04 dedicated servers that I mentioned on Alliedmodders). That of course could mean that Valve hasn't seen the project yet....but I doubt that.

I guess only time will tell.

**Comment by 404UserNotFound** [comment]
To finish, before I go get some sleep, we'd love to get in touch with Valve just to show them we mean no harm with our project, and to get their permission/blessing to continue working on this.

The TF2 community in general seems to love what we're doing, and we're hoping Valve would be fans as well (usage of leaked source code aside). We ourselves are fans of Valve and their work, that's why we're working on this project.

And for this project die...I don't want to see that happen.

**Comment by linouxis9** [comment]
Do you have some news from Valve regarding this point ?

**Comment by 404UserNotFound** [comment]
I'm no longer on the dev team, so I can speak freely now. When this issue was first submitted, the dev Slack chat was basically "oh shit we're so fucked", then Daniel came out with "fuck that Kyle Sanderson guy, why is he bothering us, I want to tell him to fuck off." (not exact words).

The only thing that can be done right now is for Valve to file a DMCA claim on this Git. Further reading:

> Hi there,
> 
> A great first step is to try contacting the user directly; in many cases, this type of thing turns out to be a misunderstanding that can be resolved by reaching out to them. If that's not possible, we follow the DMCA Takedown Policy.
> 
> If you are the copyright holder and want to have the content removed, please file a DMCA takedown with us via email. Instructions, examples, and policy details are available here:
> 
> https://help.github.com/articles/dmca-takedown-policy
> 
> If you are not the copyright holder, please contact the owner about this matter and have them file a DMCA takedown with us.
> 
> Thanks,
> Andrew

So basically, nothing can be done about publishing stolen code on this Git unless Valve steps in and files a DMCA claim.

**Comment by danielmm8888** [comment]
You're missing a couple of points here.

1. We've already emailed numerous people at valve numerous times, and we've yet to get a response by them.
There is absolutely no way they are not aware of this project.

2. We have already talked with one of the lead developers of SourceMod (asherkin) about you (since you publicly promised that you'd try and make a fight between them and us in a group announcement) and this mod, and he has assured us that no drama has or will be spilled between us and them because of you.

**Comment by 404UserNotFound** [comment]
> We have already talked with one of the lead developers of SourceMod (asherkin) about you (since you publicly promised that you'd try and make a fight between them and us in a group announcement) and this mod, and he has assured us that no drama has or will be spilled between us and them because of you.

Well looks like I have to contact Asherkin again and straighten things out.

**Comment by danielmm8888** [comment]
>I never once insinuated that I was looking to start up issues between AlliedMods 

Here is your full announcement:
http://steamcommunity.com/groups/gettheheckout#announcements/detail/137703207752402000
and a full page screenshot of said announcement in case you decide to edit anything on the page
http://i.imgur.com/F533Zo9.jpg

And here's the paragraph from your announcement:
>The Github situation (whereby AlliedMods user and plugin approver Kyle Sanderson called us out on the fact that we were republishing leaked/stolen source code and Danielmm8888's response was "this guy is fucking annoying, why won't he just fuck off") 

And here is your comment from that exact same announcement:
>Oh and I just now PMed KyleS on AlliedMods and explained to him the whole situation and discussions we had in Slack about him calling us out on the stolen source code. He now knows everything.


>Also regardless, the code is still stolen. That's the main problem here, and you still haven't addressed that problem aside from going "na na na boo boo i can't hear you" and/or "fuck that kyle sanderson guy"

We already have addressed it already. I don't understand what you're looking for exactly.

**Comment by 404UserNotFound** [comment]
I PMed Kyle to explain what our private stance on this issue was. Nowhere did I suggest I was starting issues up. You're pulling things out of thin air.

**Comment by 404UserNotFound** [comment]
Done, group closed. But I'll restate this bit in case it hasn't fully penetrated your skull into whatever tiny amount of grey matter is between your ears; I PMed Kyle to explain to him what our stance on this issue report he created was. Nowhere in the announcement or comment did I explicitly state or infer that I was looking to cause issues between the two communities.

Next time you make shit up, at least get some valid fucking proof to support it, or at least fucking photoshop some. Jesus christ.

**Comment by Iamgoofball** [comment]
# M-M-M-MUH DRAAAAMMMMMMAAAAAAAAAAAAAAAAASSSSSSSSSSSSSSSSSSSSSSSSSSS

# OH MY GOD CAN YOU WHINY BABY MANCHILDREN STOP STIRRING UP SHIT OVER A GODDAMN VIDEO GAME HOLY SHIT IT ISN'T THE FUCKING OLYMPICS

**Comment by Iamgoofball** [comment]
Okay, serious response time:

It's just a game. We don't need to be so angry over it. Just listen to the mst3k mantra, go to pornhub and beat it or something, I dunno. Buy a milkshake, go into the backyard, relax on a lawn chair, and think about how awesome life is.

**Comment by KyleSanderson** [comment]
It should go without saying, but my actions here and in other communities do not represent AM.

**Comment by 404UserNotFound** [comment]
Here it is. Eric S. finally replied to my email from April 7th.

![Valve's OK](http://i.imgur.com/PTMsrl7.png)

**Comment by KyleSanderson** [comment]
Perfect!

**Comment by 404UserNotFound** [comment]
There was something I forgot to mention....what was it again....oh yeah, that picture was a fake. Here's the original email:

![From Valve](http://i.imgur.com/t87zxk0.png)
